pages/api/auth/[...nextauth].ts

```diff
@@ -15,10 +15,36 @@ export const authOptions: AuthOptions = {
       }
       return false;
     },
-    async jwt({ token, user }) {
-      if (user && user?.accessToken) {
-        return { ...token, user };
+    async jwt({ token }) {
+      if (token?.exp && Date.now() < token.exp * 1000) {
+        try {
+          const response = await fetch(
+            `${API_BASE_URL}/dj-rest-auth/token/refresh/`,
+            {
+              headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+              body: new URLSearchParams({
+                refresh: token.user.refreshToken,
+              }),
+              method: 'POST',
+            }
+          );
+
+          const tokens = await response.json();
```

Comment:
I think this await is not necessary, you already have the await in line 21 and tokens is going to wait until response is resolved.

Reply:
Apparently `response.json()` returns a promise, it doesn't work if I don't await it.